lms/djangoapps/grades/models.py

```diff
@@ -828,3 +855,17 @@ def _prepare_override_params(subsection_grade_model, override_data):
                 getattr(subsection_grade_model, field_name)
             )
         return cleaned_data
+
+    @classmethod
+    def clear_override(cls, user_id, course_key):
```

Comment:
So, it looks like the foreign key relationship between Subsection Overrides and Subsection Grades is 
`grade = models.OneToOneField(PersistentSubsectionGrade, related_name='override', on_delete=models.CASCADE)`

what `on_delete=models.CASCADE` means is that if a subsection grade is deleted, any subsection grade overrides that refer to it are automatically deleted as well.

That's actually what the second return value from delete() is (the one we have been ignoring)
 from the docs:
> Performs an SQL delete query on all rows in the [QuerySet](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#django.db.models.query.QuerySet) **and returns the number of objects deleted and a dictionary with the number of deletions per object type.**

> Delete all the entries belonging to this Blog.
 Entry.objects.filter(blog=b).delete()
 (4, {'blog.Entry': 2, 'blog.Entry_authors': 2})

So in this case, when we delete all of the subsection grades for a learner, we'll also be deleting the grade overrides as well, automatically.

That means, for this model, all we really have to worry about is cache invalidation. I'd suggest turning this method into something like a `clear_prefetched_overrides_for_learner`, and just pop the cache entry for a specified (user_id, course_key)

Then, when we delete subsection grades, we can check the second return value to see if there were any subsection overrides, and if there were, call the method to clear the cache for the learner and course.

Reply:
Gotcha...missed the relationship part 🤦